src/routes/MetaDetails/StreamsList/StreamsList.js

```diff
@@ -64,96 +64,106 @@ const StreamsList = ({ className, video, ...props }) => {
-                                        <Stream
-                                            key={index}
-                                            videoId={video?.id}
-                                            videoReleased={video?.released}
-                                            addonName={stream.addonName}
-                                            name={stream.name}
-                                            description={stream.description}
-                                            thumbnail={stream.thumbnail}
-                                            progress={stream.progress}
-                                            deepLinks={stream.deepLinks}
-                                            onClick={stream.onClick}
-                                        />
-                                    ))}
+                                <Button
+                                    className={classnames(
+                                        styles['button-container'],
+                                        styles['back-button-container']
+                                    )}
+                                    tabIndex={-1}
+                                    onClick={backButtonOnClick}
+                                >
+                                    <Icon className={styles['icon']} name={'chevron-back'} />
+                                </Button>
+                                <div className={styles['episode-title']}>
+                                    {`S${video?.season}E${video?.episode} ${video?.title}`}
                                 </div>
                             </React.Fragment>
-            }
-            <Button className={styles['install-button-container']} title={t('ADDON_CATALOGUE_MORE')} href={'#/addons'}>
+                        ) : null}
+                        {Object.keys(streamsByAddon).length > 1 ? (
+                            <Multiselect
+                                {...selectableOptions}
+                                className={styles['select-input-container']}
+                            />
+                        ) : null}
+                    </div>
+                    <div className={styles['streams-container']}>
+                        {filteredStreams.map((stream, index) => (
+                            <Stream
+                                key={index}
+                                videoId={video?.id}
+                                videoReleased={video?.released}
+                                addonName={stream.addonName}
+                                name={stream.name}
+                                description={stream.description}
+                                thumbnail={stream.thumbnail}
+                                progress={stream.progress}
+                                deepLinks={stream.deepLinks}
+                                onClick={stream.onClick}
+                                infoHash={stream.infoHash}
```

Comment:
Instead of passing the stream infoHash to the Stream component
It could use the `deepLinks.externalPlayer.download` inside the Stream component, this ways it would return the full magnet link when available and also allow to copy the url of other streams